### PR TITLE
Fix BMD Audio + Add Volume Button

### LINKED ISSUE #2117: VxMark: no audio in headphones
## Description
There is no audio when using headphones on a locked down production VxMark [using this image](https://s3.console.aws.amazon.com/s3/object/votingworks-machine-images?region=us-west-2&prefix=2022-07-nh/2022-07-08-vxmark.img.lz4) for the NH demo.

### PR DESCRIPTION
**Pull request by adghayes**
## Overview
Closes #2117 and makes an initial pass at #1484. Split into three commits:
1. Updates `KioskBrowser` interfaces to include the `speak` APIs introduced in [votingworks/kiosk-broswer#130](https://github.com/votingworks/kiosk-browser/pull/130)
2. Adds `KioskTextToSpeech` class which uses the new `kiosk-browser` APIs, and is active when `window.kiosk` is present. This fixes BMD audio and closes #2117.
3. Adds a preliminary volume control feature. The single volume button on the audio controller will change the volume up and down, cycling. An initial pass at #1484.

I don't have the volumes mapped to decibel levels. I'm yet quite sure how to do that, or if the choice of headphones affects that. But the VVSG range is, I think, an absolute maximum and minimum - not the maximum and minimum we have to use. We can't go outside of the range they specify. Because the VVSG maximum - 100dB - is equivalent to a jet plane taking off or a jackhammer. Perhaps I am misunderstanding something. 

In any case, for this first pass, I think we should just settle on reasonable defaults. In this implementation, there are 10 possible audio levels (1 of 10 to 10 of 10) mapping to an underlying audio setting of 28% to 100% volume. That underlying range is what roughly I think a user might need, and it led to pretty-ish numbers.

## Testing Plan 
- Unit tests for `KioskTextToSpeech`
- Test for the keypress detection in `app.test.tsx`

## Checklist
- [ ] ~~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~~
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: frontends/bmd/src/app.tsx [hunk 1]
```diff
@@ -76,11 +76,12 @@ export function App({
     }
   }, [internalHardware, screenReader]);
 
-  /* istanbul ignore next - need to figure out how to test this */
-  const onKeyPress = useCallback(
+  const onKeyDown = useCallback(
```
Comment (adghayes) [inline frontends/bmd/src/app.tsx:79]: Have to use KeyPress to detect keys that do not produce a character, like `F17` from the Accessible Controller.

File: frontends/bmd/src/app.test.tsx [hunk 2]
```diff
@@ -72,6 +74,38 @@ it('uses kiosk storage when in kiosk-browser', async () => {
...
+  // check that 'r' toggles the screen reader
+  expect(screenReader.toggle).toHaveBeenCalledTimes(0);
+  fireEvent.keyDown(screen.getByText('Device Not Configured'), { key: 'r' });
+  await waitFor(() => {
+    expect(screenReader.toggle).toHaveBeenCalledTimes(1);
+  });
+
+  // check that 'F17' changes volume
+  expect(screenReader.changeVolume).toHaveBeenCalledTimes(0);
+  fireEvent.keyDown(screen.getByText('Device Not Configured'), { key: 'F17' });
```
Comment (benadida) [inline frontends/bmd/src/app.test.tsx:105]: If there's any way to not depend on the a particular piece of text here that is irrelevant to the action, that will make this test a little bit more robust if we choose to change the layout / wording for the "Device Not Configured" screen. Maybe add something with a `test-id`, and use `getByTestId` ?

Not strictly required to merge, just a suggestion.

File: frontends/bmd/src/app.tsx (hunk 1 above)
Comment (benadida) [reply frontends/bmd/src/app.tsx:79]: are we sure this change doesn't affect behavior of the accessible controller for other actions?

**Review by benadida — APPROVED** [review]
looks good! One small suggestion, take it or leave it, and one double check that the key event change didn't screw anything else up.

File: frontends/bmd/src/app.test.tsx (hunk 2 above)
Comment (adghayes) [reply frontends/bmd/src/app.test.tsx:105]: Good call, just found I can use `getByRole('main')` which is a degree more resilient.

**Comment by adghayes** [comment]
I am almost certain that the switch the `keyDown` does not affect any behavior. First of all, [`keyPress` has been deprecated ](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event) in favor of `keyDown` and `keyUp`. The behavior of the accessible controller's other buttons is not handled by our own custom events, but by the browser itself, so it shouldn't affect that.

File: frontends/bmd/src/utils/ScreenReader/kiosk_text_to_speech.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  constructor() {
+    assert(window.kiosk, 'KioskTextToSpeech requires window.kiosk');
+  }
+
+  /**
+   * Directly triggers speech of text. Resolves when speaking is finished
+   * or cancelled.
+   */
+  async speak(text: string, { now = true }: SpeakOptions = {}): Promise<void> {
+    assert(now, 'method "speak"  with "options.now = false" is not supported');
```
Comment (eventualbuddha) [inline frontends/bmd/src/utils/ScreenReader/kiosk_text_to_speech.ts:35]: As @benadida mentioned in his earlier comment on this, I don't think we ever actually use `now = false`, so maybe we just strip it out altogether?

**Pull request merged**